### PR TITLE
MTDSA-14344: TYS changes for Ignore State Benefit

### PR DESCRIPTION
**Pull request by GazalNebHMRC**

### CODE REVIEW

File: app/config/FeatureSwitches.scala [hunk 1]
```diff
@@ -28,19 +28,19 @@ case class FeatureSwitches(featureSwitchConfig: Configuration) {
...
-      enabled      <- featureSwitchConfig.getOptional[Boolean](s"version-$validVersion.enabled")
+      enabled <- featureSwitchConfig.getOptional[Boolean](s"version-$validVersion.enabled")
     } yield enabled
 
     enabled.getOrElse(false)
   }
 
   val isTaxYearNotEndedRuleEnabled: Boolean = isEnabled("taxYearNotEndedRule.enabled")
-  val isTaxYearSpecificApiEnabled: Boolean  = isEnabled("tys-api.enabled")
+  val isTaxYearSpecificApiEnabled: Boolean = isEnabled("tys-api.enabled")
```
Comment (SamuelJacobsHMRC) [inline app/config/FeatureSwitches.scala:43]: I may be wrong but it looks like the scala formatting has been removed and that is the only change in this file? Can we format this again to be sure

File: test/v1/connectors/IgnoreBenefitConnectorSpec.scala [hunk 2]
```diff
@@ -27,42 +25,52 @@ import scala.concurrent.Future
...
-  val taxYear: String               = "2019-20"
-  val benefitId: String             = "123e4567-e89b-12d3-a456-426614174000"
-  val request: IgnoreBenefitRequest = IgnoreBenefitRequest(Nino(nino), TaxYear.fromMtd(taxYear), benefitId)
+  "IgnoreBenefitConnector" should {
+    "return the expected response for a non-TYS request" when {
+      "a valid request is made" in new IfsTest with Test {
+        def taxYear: TaxYear = TaxYear.fromMtd("2019-20")
 
-  class Test extends MockHttpClient with MockAppConfig {
+        private val outcome = Right(ResponseWrapper(correlationId, ()))
```
Comment (SamuelJacobsHMRC) [inline test/v1/connectors/IgnoreBenefitConnectorSpec.scala:33]: Can we add the type here

File: test/v1/connectors/IgnoreBenefitConnectorSpec.scala [hunk 3]
```diff
@@ -27,42 +25,52 @@ import scala.concurrent.Future
...
-      "return a successful response" in new Test {
-        private val outcome = Right(ResponseWrapper(correlationId, ()))
+        result shouldBe outcome
+      }
+
+      "return the expected response for a TYS request" when {
+        "a valid request is made" in new TysIfsTest with Test {
+          def taxYear: TaxYear = TaxYear.fromMtd("2023-24")
+
+          private val outcome = Right(ResponseWrapper(correlationId, ()))
```
Comment (SamuelJacobsHMRC) [inline test/v1/connectors/IgnoreBenefitConnectorSpec.scala:48]: can we add the type here

File: app/config/FeatureSwitches.scala (hunk 1 above)
Comment (GazalNebHMRC) [reply app/config/FeatureSwitches.scala:43]: This is just how my scala formatter is working. I don't think I can change that.

File: test/v1/connectors/IgnoreBenefitConnectorSpec.scala (hunk 2 above)
Comment (GazalNebHMRC) [reply test/v1/connectors/IgnoreBenefitConnectorSpec.scala]: Updated, thanks.

**Review by SamuelJacobsHMRC — APPROVED** [review]

**Pull request merged**